tests/versioned/apollo-server-express/segments.test.js.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+
+const utils = require('@newrelic/test-utilities')
+utils.assert.extendTap(tap)
+
+const { getTypeDefs, resolvers } = require('../data-definitions')
+const { createSegmentsTests } = require('../express-segments-tests')
+
+tap.test('apollo-server-express: segments', (t) => {
+  t.autoend()
+
+  let server = null
+  let expressServer = null
+  let serverUrl = null
+  let helper = null
+
+  t.beforeEach((done) => {
+    // load default instrumentation. express being critical
+    helper = utils.TestAgent.makeInstrumented()
+    const createPlugin = require('../../../lib/create-plugin')
+    const nrApi = helper.getAgentApi()
+
+    // TODO: eventually use proper function for instrumenting and not .shim
+    const plugin = createPlugin(nrApi.shim)
+
+    const express = require('express')
+
+    // Do after instrumentation to ensure express isn't loaded too soon.
+    const { ApolloServer, gql } = require('apollo-server-express')
+    server = new ApolloServer({
+      typeDefs: getTypeDefs(gql),
+      resolvers,
+      plugins: [plugin]
+    })
+
+    const app = express()
+    server.applyMiddleware({ app })
+
+    expressServer = app.listen(0, () => {
+      serverUrl = `http://localhost:${expressServer.address().port}${server.graphqlPath}`
+
+      t.context.helper = helper
+      t.context.serverUrl = serverUrl
+      done()
+    })
+  })
+
+  t.afterEach((done) => {
+    expressServer && expressServer.close()
+    server && server.stop()
+
+    helper.unload()
+    server = null
+    serverUrl = null
+    helper = null
+
+    clearCachedModules(['express', 'apollo-server-express'], () => {
```

Comment:
Clearing modules is critical between test runs to get consistent results. Otherwise, express won't necessarily get instrumented, etc.

Reply:
til